Code/GraphMol/SmilesParse/CXSmilesOps.cpp

```diff
@@ -266,7 +266,7 @@ bool parse_it(Iterator &first, Iterator last, RDKit::RWMol &mol) {
       } else {
         if (!parse_atom_labels(first, last, mol)) return false;
       }
-    } else if ((first + 9) < last &&
+    } else if (std::distance(first, last) > 9 &&
```

Comment:
I agree with Brian: it makes sense to fix this instance of the bug at the same time.
There's another call that could fail this way at line 53.